var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -166,12 +177,12 @@ class Cmake(Package):
     depends_on('libuv@1.10.0:1.10.99',  when='@3.11.0:3.11.99~ownlibs')
     depends_on('libuv@1.10.0:',  when='@3.12.0:~ownlibs')
     depends_on('rhash',          when='@3.8.0:~ownlibs')
-    depends_on('qt',             when='+qt')
-    depends_on('python@2.7.11:', when='+doc', type='build')
-    depends_on('py-sphinx',      when='+doc', type='build')
-    depends_on('openssl', when='+openssl')
-    depends_on('openssl@:1.0.99', when='@:3.6.9+openssl')
-    depends_on('ncurses',        when='+ncurses')
+    depends_on('qt',             when='+qt~ownlibs')
+    depends_on('python@2.7.11:', when='+doc~ownlibs', type='build')
+    depends_on('py-sphinx',      when='+doc~ownlibs', type='build')
+    depends_on('openssl', when='+openssl~ownlibs')
```

Comment:
This is incorrect... If you want ssl support for say FetchContent in CMake, and you use +ownlibs, it'll bootstrap curl for you, but CMake won't bootstrap curl's dependency openssl for you, likely because a system openssl is less vulnerable. So you really need `+openssl+ownlibs` here. Notice that CMake has no direct dependency on openssl.

Reply:
See also: https://github.com/spack/spack/pull/25559

Reply:
I'll respond to both comments here, so as not to clutter this PR. CMake does require `openssl` to bootstrap with curl, but `openssl` requires ZLib, which requires CMake... and Spack at the moment AFAIK (please correct me if I'm wrong) does not support a package appearing twice in the spec, and the docs state that a package such as `cmake-*` should not exist alongside a package named `cmake`. This means that CMake cannot bootstrap with `openssl` support. Additionally I agree, pulling out the ssl stanza entirely is IMO, the wrong approach. A (notedly circuitous) solution is to install cmake without openssl, then install zlib, and then uninstall the cmake with the `+ownlibs` variant, and install an edition with `openssl` enabled, now that zlib is available.

Not really ideal, in a perfect world Spack would offer more robust bootstrap capabilities, but I can confirm the above works on my local system.

Reply:
I personally prefer your 'positive' approach to enabling the `openssl` dependencies.

Reply:
Well, this change disables openssl by force, also for non-windows, which can't be right